### PR TITLE
Three js buffer geometry utils and drag controls

### PR DESCRIPTION
**Pull request by ethankay**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [NA (per existing tests to types)] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: 
DragControls: https://github.com/mrdoob/three.js/blob/dev/examples/js/controls/DragControls.js
BufferGeometryUtils: https://threejs.org/docs/#examples/BufferGeometryUtils
- [NA] Increase the version number in the header if appropriate.
- [NA] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

/cc @alexeagle

### CODE REVIEW

File: types/three/three-core.d.ts [hunk 1]
```diff
@@ -1447,6 +1447,15 @@ export class Geometry extends EventDispatcher {
     dispatchEvent(event: { type: string; [attachment: string]: any; }): void;
 }
 
+/**
+ * @see <a href="https://github.com/mrdoob/three.js/blob/master/examples/js/BufferGeometryUtils.js">examples/js/BufferGeometryUtils.js</a>
+ */
+export namespace BufferGeometryUtils {
+    export function mergeBufferGeometries(geometries: any[]): BufferGeometry;
```
Comment (dhritzkiv) [inline types/three/three-core.d.ts]: Is this an array of `any`?

File: types/three/three-core.d.ts [hunk 2]
```diff
@@ -1447,6 +1447,15 @@ export class Geometry extends EventDispatcher {
...
+    export function computeTangents(geometry: BufferGeometry): null;
+    export function mergeBufferAttributes(attributes: any[]): BufferAttribute;
```
Comment (dhritzkiv) [inline types/three/three-core.d.ts]: Is this an array of `any`?

File: types/three/three-core.d.ts [hunk 3]
```diff
@@ -1447,6 +1447,15 @@ export class Geometry extends EventDispatcher {
     dispatchEvent(event: { type: string; [attachment: string]: any; }): void;
 }
 
+/**
+ * @see <a href="https://github.com/mrdoob/three.js/blob/master/examples/js/BufferGeometryUtils.js">examples/js/BufferGeometryUtils.js</a>
```
Comment (dhritzkiv) [inline types/three/three-core.d.ts]: I don't think it's necessary to wrap the link in an anchor tag. See [\@see documentation](http://usejsdoc.org/tags-see.html)

File: types/three/three-dragcontrols.d.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import {Camera} from './three-core';
+
+export class DragControls {
+    constructor(objects: any, camera: Camera, domElement?: HTMLElement);
```
Comment (dhritzkiv) [inline types/three/three-dragcontrols.d.ts]: Is `objects` supposed to be `any`?

File: types/three/three-core.d.ts (hunk 1 above)
Comment (XanderLuciano) [reply types/three/three-core.d.ts]: That is an array of `any`, but it (probably) should be an array of `BufferGeometry` as stated here: https://threejs.org/docs/#examples/BufferGeometryUtils

```typescript
export function mergeBufferGeometries(geometries: BufferGeometry[]): BufferGeometry;
```

File: types/three/three-core.d.ts (hunk 2 above)
Comment (XanderLuciano) [reply types/three/three-core.d.ts]: Again, should be an array of `BufferAttribute`s

```typescript
export function mergeBufferAttributes(attributes: BufferAttribute[]): BufferAttribute;
```

File: types/three/three-dragcontrols.d.ts (hunk 4 above)
Comment (XanderLuciano) [reply types/three/three-dragcontrols.d.ts]: Not 100% on this one, but I think `objects` expects `Object3d`

File: types/three/three-core.d.ts (hunk 1 above)
Comment (dhritzkiv) [reply types/three/three-core.d.ts]: Great! Could you make those changes?

File: types/three/three-core.d.ts (hunk 3 above)
Comment (elk941) [reply types/three/three-core.d.ts]: Much of the rest of the three-core.d.ts classes seemed to follow the see convention (that's why I went with that at first). Any reason in particular we want to deviate in this case?

File: types/three/three-core.d.ts (hunk 2 above)
Comment (elk941) [reply types/three/three-core.d.ts]: Done.

File: types/three/three-dragcontrols.d.ts (hunk 4 above)
Comment (elk941) [reply types/three/three-dragcontrols.d.ts]: Looking into the raycaster types (used by DragControls), it looks like this should be an array, so `Object3D[]`. I'll make the update.

File: types/three/three-core.d.ts (hunk 1 above)
Comment (elk941) [reply types/three/three-core.d.ts]: Done.

File: types/three/three-core.d.ts (hunk 3 above)
Comment (dhritzkiv) [reply types/three/three-core.d.ts]: I could be wrong, but I suspect the other instances are incorrectly formatted. This is a minor nitpick. I'll let the maintainers decide.

**Review by dhritzkiv — APPROVED** [review]
All seems fine minus nitpick about `@see` usage

File: types/three/three-core.d.ts (hunk 3 above)
Comment (elk941) [reply types/three/three-core.d.ts]: Sounds good. Happy to change it to whatever format is best if requested.

**Pull request merged**